cmd/scaffold.go

```diff
@@ -455,14 +459,17 @@ func (fnb *FlowNodeBuilder) initState() {
 		// not use a global variable for chain ID anymore, but rely on the protocol
 		// state as final authority on what the chain ID is
 		// => https://github.com/dapperlabs/flow-go/issues/4167
-		fnb.RootChainID = stateRoot.Block().Header.ChainID
+		fnb.RootChainID = rootBlock.ChainID
 
 		// load the root QC data from bootstrap files
+		// TODO get from protocol state
 		fnb.RootQC, err = loadRootQC(fnb.BaseConfig.BootstrapDir)
 		fnb.MustNot(err).Msg("could not load root QC")
 
-		fnb.RootResult = stateRoot.Result()
-		fnb.RootSeal = stateRoot.Seal()
+		fnb.RootResult, err = state.AtBlockID(rootBlock.ID()).LatestResult()
+		fnb.MustNot(err).Msg("could not get root result")
+		fnb.RootSeal, err = state.AtBlockID(rootBlock.ID()).LatestSeal()
+		fnb.MustNot(err).Msg("could not get root seal")
```

Comment:
The data returned by `LatestResult` and `LatestSeal` are related.

Do we need them individually? Otherwise, I would suggest to make a `LatestSealedResult` function that returns both the seal and the sealed result, since I guess `LatestResult`  anyway need to query the latest seal. And this also eliminates the case where the returned seal and the result doesn't match.

Reply:
👍 Good idea, will apply in https://github.com/onflow/flow-go/pull/374